### DISCUSSION TITLE
Is it possible to perform dynamic page routing using StreamSync?

### DISCUSSION THREAD

**Discussion by PankajMoolrajani** [writer/writer-framework#23] [General]
We have an application with hundreds of pages, and the content for these pages is stored in a database. When a user clicks on a link, they will be redirected to a new page. However, this new page is not individually coded as a separate file. Instead, a template is established, for instance example.com/<new-page-path>, and the content for this page is retrieved from the database.

**Comment by ramedina86** [comment]
Absolutely, it's really simple. I'm sharing a zip with an example. It's less than 10 lines of code, excluding some hardcoded data. Decompress it and you can do `streamsync edit pagetest` from the containing folder.

You'd create two pages (e.g. main and detail), use a _Repeater_ component to list items, load the detail state and then switch pages.

```py
def handle_product_selection(state, context):
    state["detail"] = products[context["itemId"]] # Load the selected item
    state.set_page("detail_page") # Switch active page

ss.init_state({
    "products": products,
    "detail": None
})
```

If the pages need to have URLs so that they can be shared, that can be set via [page routes](https://www.streamsync.cloud/page-routes.html).

Let me know if you need any further tips, I'd be happy to help.

[pagetest.zip](https://github.com/ramedina86/streamsync/files/11436094/pagetest.zip)
